error compiling/linking with Centos6 openssl

Because of patent concerns, RedHat (and by extension, Centos) remove `ec.h` and `ecdsa.h` from the distribution of openssl.  This causes problems in building hogweed. 

See https://bugzilla.redhat.com/show_bug.cgi?id=319901

``` bash
...
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:51:24: error: openssl/ec.h: No such file or directory
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:52:27: error: openssl/ecdsa.h: No such file or directory
gcc -g -O2 -ggdb3 -Wno-pointer-sign -Wall -W   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   -Wpointer-arith -Wbad-function-cast -Wnested-externs -L.. -L/home/kmsquire/.julia/v0.2/Nettle/deps/usr/lib -L/home/kmsquire/src/julia/usr//lib -Wl,-rpath -Wl,/home/kmsquire/.julia/v0.2/Nettle/deps/usr/lib base16enc.o io.o \
    -lnettle -lgmp -lgmp  -o base16enc
gcc -g -O2 -ggdb3 -Wno-pointer-sign -Wall -W   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   -Wpointer-arith -Wbad-function-cast -Wnested-externs -L.. -L/home/kmsquire/.julia/v0.2/Nettle/deps/usr/lib -L/home/kmsquire/src/julia/usr//lib -Wl,-rpath -Wl,/home/kmsquire/.julia/v0.2/Nettle/deps/usr/lib base16dec.o io.o \
    -lnettle -lgmp -lgmp  -o base16dec
gcc -g -O2 -ggdb3 -Wno-pointer-sign -Wall -W   -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   -Wpointer-arith -Wbad-function-cast -Wnested-externs -L.. -L/home/kmsquire/.julia/v0.2/Nettle/deps/usr/lib -L/home/kmsquire/src/julia/usr//lib -Wl,-rpath -Wl,/home/kmsquire/.julia/v0.2/Nettle/deps/usr/lib base64enc.o io.o \
    -lnettle -lgmp -lgmp  -o base64enc
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:497: error: expected specifier-qualifier-list before 'EC_KEY'
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c: In function 'bench_openssl_init':
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:519: error: 'struct openssl_ctx' has no member named 'key'
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:519: warning: implicit declaration of function 'EC_KEY_new_by_curve_name'
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:519: warning: nested extern declaration of 'EC_KEY_new_by_curve_name'
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:520: error: 'struct openssl_ctx' has no member named 'digest_length'
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:521: error: 'struct openssl_ctx' has no member named 'digest'
/home/kmsquire/.julia/v0.2/Nettle/deps/src/nettle-2.7.1/examples/hogweed-benchmark.c:531: error: 'struct openssl_ctx' has no member named 'key'
...
```

--- a/deps/build.jl
+++ b/deps/build.jl
@@ -31,7 +31,12 @@ env = {"HOGWEED_LIBS" => "-L$(libdirs[1]) -L$(BinDeps.libdir(nettle)) -lhogweed 
        "NETTLE_LIBS" => "-L$(libdirs[1]) -L$(BinDeps.libdir(nettle)) -lnettle -lgmp",
        "LIBS" => "-lgmp ", "LD_LIBRARY_PATH" => join([libdirs[1];BinDeps.libdir(nettle)],":")}
 
-provides( Sources, URI("http://www.lysator.liu.se/~nisse/archive/nettle-2.7.tar.gz"), nettle )
-provides( BuildProcess, Autotools(lib_dirs = libdirs, include_dirs = includedirs, env = env), nettle )
+provides( Sources, URI("http://www.lysator.liu.se/~nisse/archive/nettle-2.7.1.tar.gz"), nettle )
+provides( BuildProcess,
+          Autotools(lib_dirs = libdirs,
+                    include_dirs = includedirs,
+                    env = env,
+                    configure_options = ["--disable-openssl", "--libdir=$(BinDeps.libdir(nettle))"]),
+          nettle )
 
 @BinDeps.install

